export_dagmc_cmd/DAGMCExportCommand.cpp

```diff
@@ -0,0 +1,825 @@
+
+  moab::EntityHandle set = 0;
+  rval = mdbImpl->tag_set_data(faceting_tol_tag, &set, 1, &faceting_tol);
+  CHK_MB_ERR_RET_MB("Error setting faceting tolerance tag",rval);
+
+  // read parsed command for normal tolerance
+  data.get_value("normal_tolerance",norm_tol);
+  message << "Setting normal tolerance to " << norm_tol << std::endl;
+  
+  // read parsed command for verbosity
+  verbose_warnings = data.find_keyword("verbose");
+  fatal_on_curves = data.find_keyword("fatal_on_curves");
+
+  if (verbose_warnings && fatal_on_curves)
+    message << "This export will fail if curves fail to facet" << std::endl;
+
+
+}
+
+moab::ErrorCode DAGMCExportCommand::create_tags()
+{
+  moab::ErrorCode rval;
+
+
+  // get some tag handles
+  int negone = -1, zero = 0 /*, negonearr[] = {-1, -1, -1, -1}*/;
+  rval = mdbImpl->tag_get_handle(GEOM_DIMENSION_TAG_NAME, 1, moab::MB_TYPE_INTEGER,
+                                 geom_tag, moab::MB_TAG_SPARSE | moab::MB_TAG_CREAT, &negone);
+  CHK_MB_ERR_RET_MB("Error creating geom_tag",rval);
+    
+  rval = mdbImpl->tag_get_handle(GLOBAL_ID_TAG_NAME, 1, moab::MB_TYPE_INTEGER,
+                                 id_tag, moab::MB_TAG_DENSE | moab::MB_TAG_CREAT, &zero);
+  CHK_MB_ERR_RET_MB("Error creating id_tag",rval);
+  
+  rval = mdbImpl->tag_get_handle(NAME_TAG_NAME, NAME_TAG_SIZE, moab::MB_TYPE_OPAQUE,
+                                 name_tag, moab::MB_TAG_SPARSE | moab::MB_TAG_CREAT);
+  CHK_MB_ERR_RET_MB("Error creating name_tag",rval);
+  
+  rval = mdbImpl->tag_get_handle(CATEGORY_TAG_NAME, CATEGORY_TAG_SIZE, moab::MB_TYPE_OPAQUE,
+                                 category_tag, moab::MB_TAG_SPARSE | moab::MB_TAG_CREAT);
+  CHK_MB_ERR_RET_MB("Error creating category_tag",rval);
+
+  rval = mdbImpl->tag_get_handle("FACETING_TOL", 1, moab::MB_TYPE_DOUBLE, faceting_tol_tag,
+                                 moab::MB_TAG_SPARSE | moab::MB_TAG_CREAT);
+  CHK_MB_ERR_RET_MB("Error creating faceting_tol_tag",rval);
+
+  rval = mdbImpl->tag_get_handle("GEOMETRY_RESABS", 1, moab::MB_TYPE_DOUBLE, 
+                                 geometry_resabs_tag, moab::MB_TAG_SPARSE | moab::MB_TAG_CREAT);
+  CHK_MB_ERR_RET_MB("Error creating geometry_resabs_tag",rval);
+
+  moab::EntityHandle set = 0;
+  rval = mdbImpl->tag_set_data(geometry_resabs_tag, &set, 1, &GEOMETRY_RESABS);
+  CHK_MB_ERR_RET_MB("Error setting geometry_resabs_tag",rval);
+
+}
+
+void DAGMCExportCommand::teardown()
+{
+  message  << "***** Faceting Summary Information *****" << std::endl
+           << "----- Curve Fail Information -----" << std::endl
+           << "There were " << failed_curve_count << " curves that could not be faceted." << std::endl
+           << "----- Facet Fail Information -----" << std::endl
+           << "There were " << failed_surface_count << " surfaces that could not be faceted." << std::endl
+           << "***** End of Faceting Summary Information *****" << std::endl;
```

Comment:
This might be a nice opportunity to reduce user concern by altering the output message to skip a failure report if all curves/surfaces are successfully faceted.